### ISSUE TITLE
Force redrawing of the graph on NSAppearance change (release-2-3 branch)

### ISSUE THREAD

**Issue by DivineDominion** [core-plot/core-plot#418]
When the system or app appearance change between light and dark mode, this does not trigger a redraw. (macOS)

When you use adaptive colors, like `NSColor.systemBlue` or `NSColor.secondaryLabelColor`, the layer colors do indeed change when I e.g. resize the window and the hosting view with it. So the color adaptation through `CPTColor` is not a problem; axis lines and tick marks work well when resizing the view:

            let lineStyle = CPTMutableLineStyle()
            lineStyle.lineWidth = 1
            lineStyle.lineColor = CPTColor(nsColor: .secondaryLabelColor)
      
            let textStyle = CPTMutableTextStyle()
            textStyle.color = CPTColor(nsColor: .secondaryLabelColor)

            if let xAxis = axisSet.xAxis {
                xAxis.axisLineStyle = lineStyle
                xAxis.majorTickLineStyle = lineStyle
                xAxis.labelTextStyle = textStyle
            }

But I didn't find a way to invalidate the cached tick labels programmatically. Here you see what happens when I switch to dark appearance and then pinch-to-zoom the graph: new tick labels appear in the correct color, but the re-used label "400" retains its old color.

![image](https://user-images.githubusercontent.com/59080/68014446-eb512f80-fc8f-11e9-9294-b5265f0b8c2d.png)

I tried these to no avail:

- `graph.reloadData()`
- `graph.setNeedsDisplay()`
- `graph.setNeedsLayout()`
- `graph.sublayers?.forEach { $0.setNeedsDisplay(); $0.displayIfNeeded(); $0.setNeedsLayout(); $0.layoutSublayers() }`
- `graph.layoutSublayers()`
- `hostingView.updateLayer()`

This could all stem from a misunderstanding of layer drawing and invalidation :)

If others want to join in with the experiment, here's some boilerplate to test this:

```swift
internal class ViewController: NSViewController {
    @IBOutlet var hostingView: CPTGraphHostingView!

    private var appearanceChangeObservation: NSKeyValueObservation?

    override func viewDidLoad() {
        super.viewDidLoad()
        self.appearanceChangeObservation = self.view.observe(\.effectiveAppearance, options: [.new]) { [unowned self] (view, change) in
            self.updateGraphAppearance()
        }
    }

    private func updateGraphAppearance() {
        // TODO: Figure out what to put here :)
    }
}
```

**Comment by DivineDominion** [comment]
Calls to `-[CPTXYAxis setNeedsRelabel]` and `relabel` do update the tick marks and axes lines.

But e.g. clearing the `axisLabels` set does, well, remove all axis labels but won't trigger a re-creation of labels through `-[CPTAxis updateAxisLabelsAtLocations: inRange: useMajorAxisLabels:]`, although that seems to be the place where old labels are reused.

My brute-force attempt:

```swift
let axisSet = graph.axisSet as! CPTXYAxisSet
axisSet.xAxis?.setNeedsRelabel()
axisSet.xAxis?.relabel()
// This does not help:
// axisSet.xAxis?.axisLabels = nil
// axisSet.xAxis?.labelFormatterChanged
axisSet.xAxis?.updateMajorTickLabels()

axisSet.yAxis?.setNeedsRelabel()
axisSet.yAxis?.relabel()
axisSet.yAxis?.updateMajorTickLabels()
```

**Comment by eskroch** [comment]
Fixed on the `release-2.3` branch.
